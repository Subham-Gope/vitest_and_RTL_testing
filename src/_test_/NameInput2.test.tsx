// import { describe, test, expect } from "vitest";
// import { render, screen } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import NameInput2 from "../components/NameInput2";

// describe("Testing the NameInput2.tsx component", () => {
//   test("updates value of input simulating typing", async () => {
//     const { container } = render(
//       <NameInput2 message={`Testing the userEvent.click feature in RTL`} />
//     );

//     expect(container).toMatchSnapshot();

//     const input = screen.getByPlaceholderText("Subham Gope");

//     await userEvent.type(input, "Sigma Lord");
//     expect(input).toHaveValue(`Sigma Lord`);
//   });
// });
