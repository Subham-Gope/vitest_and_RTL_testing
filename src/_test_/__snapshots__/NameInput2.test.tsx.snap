// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the NameInput2.tsx component > updates value of input simulating typing 1`] = `
<div>
  <p>
    Testing the userEvent.click feature in RTL
  </p>
  <div>
    <label
      for="name"
    >
      Name:
       
      <input
        id="name"
        name="name"
        placeholder="Subham Gope"
        type="text"
        value=""
      />
       
    </label>
  </div>
</div>
`;
